java/com/jfreechartextensions/axistransformations/ExtendedNumberAxis.java

```diff
@@ -191,32 +200,37 @@ protected void selectHorizontalAutoTickUnit(Graphics2D g2, Rectangle2D dataArea,
         TickUnit unit1 = tickUnits.getCeilingTickUnit(this.getTickUnit());
 
         //if transform
-        final double unit1WidthUpper = Math.abs(this.lengthToJava2D(getRange().getUpperBound(), dataArea, edge) - this.lengthToJava2D(getRange().getUpperBound() - unit1.getSize(), dataArea, edge));
-        final double unit1WidthLower = Math.abs(this.lengthToJava2D(getRange().getLowerBound(), dataArea, edge) - this.lengthToJava2D(getRange().getLowerBound() + unit1.getSize(), dataArea, edge));
+        final double unit1WidthUpper = Math.abs(this.lengthToJava2D(Math.abs(getRange().getUpperBound()), dataArea, edge) - this.lengthToJava2D(Math.abs(getRange().getUpperBound() - unit1.getSize()),
+                dataArea,
+                edge));
+        final double unit1WidthLower = Math.abs(this.lengthToJava2D(Math.abs(getRange().getLowerBound()), dataArea, edge) - this.lengthToJava2D(Math.abs(getRange().getLowerBound() + unit1.getSize()),
```

Comment:
Is it expensive to call getRange() multiple times?  Should it just be called once and used multiple times?

Reply:
Is it expensive to call getRange() multiple times?  Should it just be called once and used multiple times?

Also, the abs calls around the range bounds seem supect.

Reply:
I think the abs() is ok here. Worth taking a second look

Reply:
I put Math.abs to avoid negative bounds affecting the length computation while transformations applied as it came across while working on this fix. Now that the range computation makes sure that invisible points won't be visible, I can take those off and check again.